### PR TITLE
added new hook before_die

### PR DESCRIPTION
**Pull request by migliozziilaria**
before die, the script calls the new hook (before_die) in oder to execute extra user's operations.

Alternatively, you could call the before_exit hook.

### CODE REVIEW

**Comment by ynuyasha** [comment]
Hi @percona-csalguero someone can take care of this PR?

**Comment by percona-csalguero** [comment]
Hi, thanks for the contribution.
Would it be possible to write a test? There are examples in the `t` directory.

Thanks

**Review by svetasmirnova — CHANGES_REQUESTED** [review]
Please add a test case as was suggested by @percona-csalguero

File: bin/pt-online-schema-change [hunk 1]
```diff
@@ -8412,7 +8412,7 @@ my $term_readkey = eval {
 
 use sigtrap 'handler', \&sig_int, 'normal-signals';
 
-
+my $plugin;
```
Comment (svetasmirnova) [inline bin/pt-online-schema-change:8484]: Why did you make this variable global? It is enough to have it defined inside the main function.

Comment (svetasmirnova) [reply bin/pt-online-schema-change:8484]: Gotcha: to use it in the sub _die.

**Review by svetasmirnova — APPROVED** [review]

**Review by EvgeniyPatlan — APPROVED** [review]

**Pull request merged**